cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -121,6 +124,12 @@ func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifest
 	}
 
 	fmt.Printf("[etcd] Wrote Static Pod manifest for a local etcd instance to %q\n", kubeadmconstants.GetStaticPodFilepath(kubeadmconstants.Etcd, manifestDir))
+
+	fmt.Println("[etcd] Waiting for the etcd cluster to be healthy")
+	if _, err := etcdClient.WaitForClusterAvailable(etcdHealthyCheckRetries, etcdHealthyCheckInterval); err != nil {
```

Comment:
I don't like this function printing
```
    [util/etcd] Attempt timed out
    [util/etcd] Waiting 5s until next retry
    [util/etcd] Attempt timed out
    [util/etcd] Waiting 5s until next retry
```
IMO those output should be removed (or converted into log messages) in order to be consistent with all the other waiters in kubeadm.

However, considering that this requires to add "This can take up to ..." in every place where `WaitForClusterAvailable` is used, this goes out of the scope of this PR, so please open an issue to track this as a todo/good first issue

Reply:
I proposed to use `klog` here too but @rosti didn't want to address this change on this PR, only changing the one potentially long with the endpoints. I agree with your point of view though @fabriziopandini.

@rosti, wdyt? Should I change this now that @fabriziopandini also raised this issue?

Reply:
Marked as resolved as per discussion with @fabriziopandini, leaving it as it was as @rosti proposed.

Reply:
I do think, that we need some sort of indication about the reason we wait another 5 seconds. This is tightly coupled with the UX of end users, that run kubeadm directly on command line. For that matter I am not a fan of klogging this. In my opinion it should go out via print.
On the other hand, we can certainly reduce the output here to say a single, more descriptive message per retry.
However, as @fabriziopandini mentioned, this will require changes in a few more places, thus it may better be done in another PR. We can file a backlog issue for now.

Reply:
i wanted to get more feedback on the 5 seconds and the interval of 20 tries.
if we can get a check faster than 5 on the average, possibly we can reduce the value?
also 20 tries is a lot. in reality, we might get the failed state much sooner.

Reply:
On my environment that is fresh everytime it took 3 retries when it was a 1 second interval, and right now with 5 seconds it retries none or once.

Without feedback on how much this takes on different environments it's hard to know (I guess it depends on the uptime of the cluster and the I/O throughput). That being said, my concern is that if we lower the timeout too much we might call it a failure when a long living one-master cluster (weeks, months?) gets a new master.

Just my 2 cents, but of course it would be great to have real data on this.

Reply:
In the last run it took 4 retries (of 5 seconds interval), this one was way off-charts and with a clean environment :(

Reply:
I also find a bit weird the output of the `kubelet-check` in the middle TBH:

```
[etcd] Waiting for the new etcd member to join the cluster. This can take up to 1m40s
I0118 12:22:43.405682    2420 etcd.go:270] attempting to see if all cluster endpoints ([https://172.28.128.56:2379 https://172.28.128.57:2379]) are available 1/20
[util/etcd] Attempt timed out
[util/etcd] Waiting 5s until next retry
I0118 12:22:53.502975    2420 etcd.go:270] attempting to see if all cluster endpoints ([https://172.28.128.56:2379 https://172.28.128.57:2379]) are available 2/20
[util/etcd] Attempt timed out
[util/etcd] Waiting 5s until next retry
I0118 12:23:03.590413    2420 etcd.go:270] attempting to see if all cluster endpoints ([https://172.28.128.56:2379 https://172.28.128.57:2379]) are available 3/20
[util/etcd] Attempt timed out
[util/etcd] Waiting 5s until next retry
[kubelet-check] Initial timeout of 40s passed.
I0118 12:23:13.615829    2420 etcd.go:270] attempting to see if all cluster endpoints ([https://172.28.128.56:2379 https://172.28.128.57:2379]) are available 4/20
I0118 12:23:13.657347    2420 join.go:625] [join] uploading currently used configuration to the cluster
```

coming from https://github.com/kubernetes/kubernetes/blob/26d58d37a71ecc698259729361484007b45a82ba/cmd/kubeadm/app/util/apiclient/wait.go#L163-L167

Reply:
i've mentioned this on slack:
> we may want to keep the overall time under 40seconds to match the kubelet timeout.
how about 2 seconds rate with 20 retries.

Reply:
@fabriziopandini @rosti please give you stamp of approval for the above comment.

Reply:
I like the 40 seconds idea, but let's keep the steps at 5 sec. Bear in mind, that we have just written out the static pod, so the kubelet needs to detect it, spin it up and for etcd to become responsive. On some systems it's easy for this to come above 2 seconds.

Reply:
ok, @rosti is voting for 5 sec / 8 retries.
@fabriziopandini ?